core/src/ibc/channel_04/manager.rs

```diff
@@ -101,4 +103,103 @@ impl<'a> Manager<'a> {
 
         Ok(key)
     }
+
+    pub fn chan_open_try(
+        &mut self,
+        order: ChannelOrder,
+        connection: Identifier,
+        channel_identifier: Identifier,
+        counterparty_channel_identifier: Identifier,
+        version: String,
+        counterparty_version: String,
+        proof_init: Bytes,
+        proof_height: u64,
+    ) -> Result<Identifier, String> {
+        let kv_store = self.ctx.get_kv_store();
+
+        let previous = kv_store.get(&channel_path(&channel_identifier, DEFAULT_PORT));
+
+        if let Some(x) = previous {
+            let channel_end: ChannelEnd = rlp::decode(&x).expect("Illformed connection end stored in the DB");
+            if channel_end.state != ChannelState::INIT
+                || channel_end.ordering != order
+                || channel_end.counterparty_port_identifier != DEFAULT_PORT
+                || channel_end.counterparty_channel_identifier != counterparty_channel_identifier
+                || channel_end.connection_hops != vec![connection.clone()]
+                || channel_end.version != version
+            {
+                return Err("There already exists ChannelEnd on which open_try() can't be conducted.".to_owned())
+            }
+        } // ok to be not found.
```

Comment:
Ditto. Comment should describe intention.

Reply:
https://www.codeconquest.com/advanced-programming-concepts/code-commenting/

Reply:
Actually that link is basically saying that  `ok to be not found` is appropriate comment here, because I clarified that it was not a mistake to miss an `else` clause.

Reply:
How about changing the comment "We can move to the `ChanOpenTry` state from `Uninitialized` state"?